### PR TITLE
fix: Only look at the current branch files' status when looking for concurrent modifications

### PR DESCRIPTION
**Pull request by carols10cents**
And, notably, ignore any new Parquet files that now appear in the catalog for this partition.

Also, don't treat this as an error yet-- only log that this happens, so we can check the logic without affecting production behavior.

### CODE REVIEW

File: compactor2/src/partition_info.rs [hunk 1]
```diff
@@ -45,27 +45,36 @@ impl PartitionInfo {
...
 }
 
+impl SavedParquetFileState {
+    pub fn existing_files_modified(&self, new: &SavedParquetFileState) -> bool {
+        let old = self;
+        let mut missing = old.ids_and_levels.difference(&new.ids_and_levels);
+
+        // If there are any files in `self`/`old` that are not present in `new`, that means some files were marked
+        // to delete by some other process.
+        missing.next().is_some()
```
Comment (NGA-TRAN) [inline compactor2/src/partition_info.rs:74]: If the level is different, it will be in this missing, right?

**Review by NGA-TRAN — APPROVED** [review]
Makes sense. I looked at the tests, too

File: compactor2/src/partition_info.rs (hunk 1 above)
Comment (carols10cents) [reply compactor2/src/partition_info.rs]: Yes, as shown in the test `changed_compaction_level_indicates_modification` in the commit I just added-- could you take a look and let me know if that's clear?

**Comment by carols10cents** [comment]
Ok, I removed the commit that changed the error filtering and added a commit that removes the error entirely, making a new component, and there's only a logging implementer of the component right now (so it's a little weird because there's no `inner` component the logging component is wrapping).

I also made the log message only print the IDs and compaction levels of the files that were in the `branch` that don't appear in the catalog set when we went to commit; this should make the log message smaller than the skipped_compaction reason that contained all of both old and new 😅 

Please review this carefully to make sure this code handles everything!!

File: compactor2/src/components/changed_files_filter/mod.rs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            .build();
+
+        // New files of any level don't affect whether the old saved state is considered modified
+        let pf_id2_level_2 = ParquetFileBuilder::new(2)
+            .with_compaction_level(CompactionLevel::Final)
+            .build();
+        let pf_id3_level_1 = ParquetFileBuilder::new(3)
+            .with_compaction_level(CompactionLevel::FileNonOverlapped)
+            .build();
+        let pf_id4_level_0 = ParquetFileBuilder::new(1)
```
Comment (NGA-TRAN) [inline compactor2/src/components/changed_files_filter/mod.rs:134]: Do you mean id = 4?

**Comment by NGA-TRAN** [comment]
It looks good to me. 
@Joe-Blount : you many want to have a look to see if you want it to only go to log for now as you suggested

**Comment by Joe-Blount** [comment]
> you many want to have a look to see if you want it to only go to log for now as you suggested

that's how it is with the current commits :+1:

File: compactor2/src/components/changed_files_filter/mod.rs (hunk 2 above)
Comment (carols10cents) [reply compactor2/src/components/changed_files_filter/mod.rs:134]: Great catch, I will fix that!

**Comment by carols10cents** [comment]
Ok, let's try this again... 

Jeff Goldblum in Independence Day saying "without the oops"

**Pull request merged**